src/src/rx_main.cpp

```diff
@@ -650,23 +652,43 @@ void ICACHE_RAM_ATTR ProcessRFPacket()
 
     uint16_t inCRC = ( ( (uint16_t)(Radio.RXdataBuffer[0] & 0b11111100) ) << 6 ) | Radio.RXdataBuffer[7];
 
-    Radio.RXdataBuffer[0] = type;
-    uint16_t calculatedCRC = ota_crc.calc(Radio.RXdataBuffer, 7, CRCInitializer);
-
-    if (inCRC != calculatedCRC)
+    // artificially inject the nonce on data packets for the CRC calculation
+    int foundNonceIndex = -1;
+    for (int index=0 ; index<3 ; index++)
     {
-        #ifndef DEBUG_SUPPRESS
+        uint8_t tryNonce = (NonceRX + nonceOffset[index]) % ExpressLRS_currAirRate_Modparams->FHSShopInterval;
+        Radio.RXdataBuffer[0] = type | (tryNonce << 2);
```

Comment:
This Isn't explicit about how many bits are available. It would be easy to mess up with a large fhss interval. Needs some defensive coding or at least a comment

Reply:
Yes, this is a bit of a problem.
In common.h it says it can be a max of 16 because we have 4 bits in the sync packet.
In the sync packet code it actually puts the entire 8bits.
On line 317 of tx_main.cpp it assumes that it can only be 2 bits.
And the actual values for the interval are all hard-coded to be 4 in the air-rate tables in common.cpp.
So, theres a bit of tidy up whichever we decide is the right option.